### PR TITLE
refactor: docker build improvements

### PR DESCRIPTION
**Pull request by dlouzan**
make docker build build the package locally instead of
depending on the npm registry published version

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM node:carbon-alpine
+FROM node:10.3-alpine
```
Comment (bufferoverflow) [inline Dockerfile]: why not `node:10-alpine` , `node:chakracore` or `node:latest` ?

**Comment by bufferoverflow** [comment]
great work, what about dumb-init is it worth to use?

File: Dockerfile (hunk 1 above)
Comment (dlouzan) [reply Dockerfile]: No preference on my side really, just used what is used in verdaccio master

Maybe @juanpicado can shared some light :-)

Comment (bufferoverflow) [reply Dockerfile]: `node:10-alpine` is the latest 10.x version, so please take this.

**Comment by dlouzan** [comment]
@bufferoverflow I find that using an init process is a docker container is a good practice to avoid any issues with signal handling. Even if we currently just start a node process, it helps with future-proffing in case we modify the startup to handle more than one process.

There's a couple of alternatives in this field:
- dumb-init
- tini
- native `--init` support in docker since version 1.13, [from the tini docs](https://github.com/krallin/tini):
  > NOTE: If you are using Docker 1.13 or greater, Tini is included in Docker itself. This includes all versions of Docker CE. To enable Tini, just pass the --init flag to docker run.

To make things a bit better right now, I'm pushing in a minute a change to this PR that actually installs dumb-init as checksummed apk, instead of downloading an executable from github. I guess that a the time this was written in verdaccio, there was still no ported apk package.

File: Dockerfile (hunk 1 above)
Comment (dlouzan) [reply Dockerfile]: 👍

**Comment by bufferoverflow** [comment]
Very nice!

**Pull request merged**